### PR TITLE
Some miscellaneous changes found while making another PR

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

These are just some minor things I thought I'd fix that I found while working on the menu bar PR.

Swapped a couple of usages of `MaterialStateProperty.all` for `const MaterialStatePropertyAll`, and fixed an anomaly in the naming of some example code data, and added a macro to the `child` docs in `ButtonStyleButton`.

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/platform_menu_bar_test.dart [hunk 1]
```diff
@@ -103,31 +103,31 @@ void main() {
                   'children': <Map<String, Object?>>[
                     <String, Object?>{
                       'id': 7,
-                      'label': 'Sub Sub Menu 100',
+                      'label': 'Sub Sub Menu 110',
```
Comment (darrenaustin) [inline packages/flutter/test/widgets/platform_menu_bar_test.dart:106]: What's the reason for the change her? Just curious.

**Review by darrenaustin — APPROVED** [review]
LGTM.

File: packages/flutter/test/widgets/platform_menu_bar_test.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/test/widgets/platform_menu_bar_test.dart:106]: It didn't match the numbering from the other ones. Basically, consistency.
